syntax = "proto3";

package catalogue_serialize;

import "map_renderer.proto";
import "transport_router.proto";

message Stop {
    string name = 1;
    uint32 id = 2;
    double lat = 3;
    double lng = 4;
    repeated Distance distances = 5;
}

message Distance {
    uint32 stop_id = 1;
    uint32 distance = 2;
}

message Bus {
    string name = 1;
    repeated uint32 id_stops = 2;
    uint32 num_stops = 3;
}

message TransportBase {
    repeated Stop stops = 1;
    repeated Bus buses = 2;
}

message TransportCatalogue {
    TransportBase base = 1;
    RenderSettings render_settings = 2;
    RoutingSettings route_settings = 3;
}
